assets/js/googlesitekit/modules/datastore/settings.test.js

```diff
@@ -38,25 +39,32 @@ describe( 'core/modules settings', () => {
 			moduleStoreName,
 			Modules.createModuleStore( slug, {
 				storeName: moduleStoreName,
-				submitChanges: () => controlReturn,
+				submitChanges: async () => {
+					await testDuringSubmissionCallback();
+					return testReturnValue;
+				},
```

Comment:
Let's update this to use a mock function instead. That way we can both make assertions that the function was called, with what arguments, but also we can change the implementation after the store has been registered.
E.g. above
```js
let submitChanges;
// ...
submitChanges = jest.fn()
```

And here just pass the function
```suggestion
				submitChanges,
```

Then in tests where we care about the return value we only need to set the internal function to return whatever we want:
```js
submitChanges.mockImplementation( () => 'foo' );
```
Or in the case of the test for `isDoingSubmitChanges`, we can simply have it return the value we care about:
```js
submitChanges.mockImplementation( ( { select } ) => select( STORE_NAME ).isDoingSubmitChanges( slug ) );
```
Then we can keep the assertion out of the callback (which might not have even failed the test anyways).

We can also add `expect( submitChanges ).toHaveBeenCalled()` as well, where this was more implicit before.

Reply:
Nice, this is a great use of jest.fn()